PyDSS/reports/voltage_metrics.py

```diff
@@ -60,9 +89,13 @@ def generate(self, output_dir):
         for scenario in self._results.scenarios:
             name = scenario.name
             data["metric_1"][name] = self._gen_metric_1(name, dfs[name])
-            data["metric_2"][name] = self._gen_metric_4(name, dfs[name], output_dir)
+            data["metric_2"][name], output = self._gen_metric_4(name, dfs[name], output_dir)
+            data["metric_4"].append(output)
             data["metric_5"][name] = self._gen_metric_5(scenario, dfs[name])
         data["metric_6"] = self._gen_metric_6(dfs)
+
+        # Note: metric 3 has to be read from the raw data. It is not duplicated
```

Comment:
Can we write  metric 3 out in the reports folder? We would want to have all metrics in the reports directory even if some are "raw", and as such, are exported as raw outputs in the exports directory. What do you think?

I know this does not sound smart from storage efficiency point of view, but looking ahead to the batch post processing of these metrics to decides on dynamic hosting capacity, it could be cumbersome and less elegant to have to read these metrics from different locations.

Reply:
I would say that the tradeoff between duplicating data and having the data in one location is not worth it. The workflow is straightforward, I hope. I'll provide a full example of how a user (Python developer) can get all voltage metrics and maybe your opinion will change.
```
from PyDSS.utils.dataframe_utils import read_dataframe
from PyDSS.pydss_results import PyDssResults

results = PyDssResults("path_to_project")
control_mode_scenario = results.scenarios[1]

# Read metrics 1, 2, 5, and 6 directly from JSON.
voltage_metrics = results.read_report("Voltage Metrics")
metric_4_filenames = voltage_metrics["metric_4"]
dfs = [read_dataframe(x) for x in filenames]

# Read all metric 3 dataframes from raw data into memory in one call.
dfs = control_mode_scenario.get_filtered_dataframes("Nodes", "VoltageMetric")

# Read metric 3 dataframes into memory one at a time.
for node_name in control_mode_scenario.list_element_names("Nodes", "VoltageMetric"):
    df = control_mode_scenario.get_dataframe("Nodes", "VoltageMetric", node_name)
    # User can convert to moving average with pandas.
```
Would it be sufficient if this example was in the documentation? Note that the steps will be different if the user chooses not to collect the data on a per-time-point basis.

Reply:
That should work. It's just bothering not to have all the metrics in one place. 
But based on the example you just gave, we can write a generic function read_metric(metric_name), when the time comes, to read these metrics so the user/developer does not have to always write different lines of codes to pull out different metrics.

Reply:
Yes, that is what I expect will happen in the code that uses these results. I updated the docstring.

Reply:
Okay. Thanks! We can move forward, then.